arm-insights/2015-04-01/swagger/insightsClient.json

```diff
@@ -0,0 +1,1123 @@
+        "category": {
+          "type": "string",
+          "description": "Gets or sets the category of the log."
+        },
+        "resultType": {
+          "type": "string",
+          "description": "Gets or sets the status of the operation."
+        },
+        "resultSignature": {
+          "type": "string",
+          "description": "Gets or sets the substatus of the operation."
+        },
+        "resultDescription": {
+          "type": "string",
+          "description": "Gets or sets the substatus of the operation."
+        },
+        "durationMs": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the duration of the operation in milliseconds."
+        },
+        "callerIpAddress": {
+          "type": "string",
+          "description": "Gets or sets the caller IP address."
+        },
+        "correlationId": {
+          "type": "string",
+          "description": "Gets or sets the correlation id of the log.  Used to group together a set of related log."
+        },
+        "identity": {
+          "type": "string",
+          "description": "Gets or sets the identity that generated the event."
+        },
+        "level": {
+          "type": "string",
+          "description": "Gets or sets the level (Informational/Warning/Error/Critical) of the event."
+        },
+        "location": {
+          "type": "string",
+          "description": "Gets or sets tThe location of the resource emitting the event."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets the collection of extended properties."
+        }
+      },
+      "description": "A log value set represents a set of log values in a time period."
+    },
+    "Log": {
+      "properties": {
+        "category": {
+          "$ref": "#/definitions/LocalizableString",
+          "description": "Gets or sets the category."
+        },
+        "startTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the start time."
+        },
+        "endTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the end time."
+        },
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LogValue"
+          },
+          "description": "Gets or sets the value of the collection."
+        }
+      },
+      "description": "The log."
+    },
+    "LogCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Log"
+          },
+          "description": "Gets or sets the value of the collection."
+        }
+      },
+      "description": "The collection of log value sets."
+    },
+    "TimeSpan": {
```

Comment:
This model definition is not required. Please use type: string and format: duration wherever TimeSpan is referenced.